app/controllers/api/v1/reports_controller.rb

```diff
@@ -38,7 +37,7 @@ def destroy
       def last
         conditions = { :host_id => Host.authorized(:view_hosts).friendly.find(params[:host_id]).try(:id) } if params[:host_id].present?
         max_id = resource_scope.where(conditions).maximum(:id)
-        @report = resource_scope.includes(:logs => [:message, :source]).find(max_id)
+        @report = resource_scope.find(max_id)
```

Comment:
can you confirm that? 1 report have many sources and messages though logs,
imho that is classicial n+1.